### PR TITLE
enable force using checked in configs + RBE on GCB tests + various fixes

### PR DESCRIPTION
**Pull request by nlopezgi**
* Adding attr that enables failing if rbe_autoconfig was not able to use checked-in configs (as requested in toolchain config launch review)

* delete outputs in external repo when output_base is used: its not possible to sanely produce artifacts that will be usable both from the @external_repo/.. and //output_base/.... So just eliminate them if output_base is used

* adding more tests that run RBE on GCB using the configs in this repo + rbe_autoconfig + RBE.

* stop using rsync (its not present in bazel containers and really not needed)

### CODE REVIEW

File: rules/rbe_repo.bzl [hunk 1]
```diff
@@ -1077,6 +1102,11 @@ def rbe_autoconfig(
         use_checked_in_confs = use_checked_in_confs,
     )
 
+    if force_checked_in_confs and not config_version:
```
Comment (smukherj1) [inline rules/rbe_repo.bzl]: Consider emitting an error if use_checked_in_confs is set to False when force_checked_in_confs is set to True

File: WORKSPACE [hunk 2]
```diff
@@ -103,7 +103,10 @@ gcs_file(
 
 load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig", "rbe_autoconfig_root")
 
-rbe_autoconfig(name = "rbe_default")
+rbe_autoconfig(
+    name = "rbe_default",
+    force_checked_in_confs = True,
```
Comment (xingao267) [inline WORKSPACE]: What about merging two flags: "force_checked_in_confs" and "use_checked_in_confs" to be just one flag, maybe "use_checked_in_confs", and the enum value can be "force" (error out if no checked in configs are available), "attempt" (try finding checked-in configs and fall back to generate configs on the fly), and "false" (always generate the configs on the fly).

File: tests/rbe_repo/generate_confs.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Steps in this file use toolchain configs produced by rbe_autoconf**
```
Comment (xingao267) [inline tests/rbe_repo/generate_confs.yaml]: Add some individual comments for each RBE build about what it's testing?

File: WORKSPACE (hunk 2 above)
Comment (nlopezgi) [reply WORKSPACE]: done

File: tests/rbe_repo/generate_confs.yaml (hunk 3 above)
Comment (nlopezgi) [reply tests/rbe_repo/generate_confs.yaml]: done

File: rules/rbe_repo.bzl (hunk 1 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: moved to using a single attr for both

**Pull request merged**